internal/expr/expr_test.go

```diff
@@ -506,6 +524,14 @@ func (s *ExprSuite) TestPrepareErrors(c *C) {
 		query:       "SELECT * AS &.* FROM t",
 		prepareArgs: []any{struct{ f int }{f: 1}},
 		err:         `cannot prepare expression: cannot use anonymous struct`,
+	}, {
+		query:       "SELECT avg(num) AS &Person.* FROM t",
+		prepareArgs: []any{Person{}},
+		err:         `cannot prepare expression: cannot use sql function with asterisk in output expression: "avg(num) AS &Person.*"`,
+	}, {
+		query:       "SELECT avg(id) AS &M.* FROM t",
+		prepareArgs: []any{sqlair.M{}},
+		err:         `cannot prepare expression: cannot use sql function with asterisk in output expression: "avg(id) AS &M.*"`,
```

Comment:
Can we add a test for the ` (fun1(...), fun2(...)) AS (&P.col1, &P.col2)` syntax?

Reply:
Can we still have a test that fails for this syntax even if the error is that the output does not have an input, even though the proper error would be better, the test is still worth adding just to see it failing.